tests/test_bkz.cpp

```diff
@@ -171,6 +177,92 @@ int test_bkz_param_pruning(ZZ_mat<ZT> &A, const int block_size, int flags = BKZ_
 }
 
 /**
+   @brief Test BKZ_DUMP_GSO for a matrix d × (d+1) integer relations matrix with bit size b (copied
+   from test_int_rel)
+
+   @param d                dimension
+   @param b                bit size
+   @param block_size       block size
+   @param float_type       floating point type to test
+   @param flags            flags to use
+
+   @return zero on success.
+ */
+template <class ZT>
+int test_int_rel_bkz_dump_gso(int d, int b, const int block_size, FloatType float_type = FT_DEFAULT,
+                              int flags = BKZ_DEFAULT | BKZ_DUMP_GSO)
+{
+  ZZ_mat<ZT> A, B;
+  A.resize(d, d + 1);
+  A.gen_intrel(b);
+  B          = A;
+  int status = 0;
+  // TODO: maybe not safe.
+  string file_bkz_dump_gso = tmpnam(nullptr);
+  status |= test_bkz_param<ZT>(B, block_size, flags, file_bkz_dump_gso);
+
+  if (status != 0)
+  {
+    cerr << "Error in test_bkz_param." << endl;
+    return status;
+  }
+
+  json js;
+  std::ifstream fs(file_bkz_dump_gso);
+  if (fs.fail())
+  {
+    cerr << "File " << file_bkz_dump_gso << " cannot be loaded." << endl;
+    return 1;
+  }
+  fs >> js;
+  int loop    = -1;
+  double time = 0.0;
+  for (auto i : js)
+  {
+    // Verify if there are as much norms as there are rows in A
+    if (A.get_rows() != (int)i["norms"].size())
+    {
+      cerr << "Array norms does not contains enough value." << endl;
```

Comment:
The array `norms` must have the same number of elements as the number of rows of A. Hope that this explains the error message. Is `Array 'norms' does not contains enough values.` correct?